File: lib/fluxsource/erasefluxsource.cc
#include "lib/core/globals.h"
#include "lib/data/fluxmap.h"
#include "lib/fluxsource/fluxsource.h"
#include "lib/fluxsource/fluxsource.pb.h"

class EraseFluxSource : public TrivialFluxSource
{
public:
    EraseFluxSource(const EraseFluxSourceProto& config)
    {
        _extraConfig.mutable_drive()->set_tracks("c0-255h0-1");
    }

    ~EraseFluxSource() {}

public:
    std::unique_ptr<const Fluxmap> readSingleFlux(int track, int side) override
    {
        return std::unique_ptr<const Fluxmap>();
    }

    void recalibrate() override {}
};

std::unique_ptr<FluxSource> FluxSource::createEraseFluxSource(
    const EraseFluxSourceProto& config)
{
    return std::make_unique<EraseFluxSource>(config);
}

File: java/com/cowlark/fluxengine/fluxsource/EraseFluxSource.java
package com.cowlark.fluxengine.fluxsource;

import com.cowlark.fluxengine.config.ConfigProto;
import com.cowlark.fluxengine.data.Fluxmap;

/**
 * A flux source which produces no flux, ported from
 * lib/fluxsource/erasefluxsource.cc.
 */
public class EraseFluxSource extends TrivialFluxSource
{
    protected ConfigProto extraConfig;

    public EraseFluxSource(EraseFluxSourceProto config)
    {
        ConfigProto.Builder builder = ConfigProto.newBuilder();
        builder.getDriveBuilder().setTracks("c0-255h0-1");
        extraConfig = builder.build();
    }

    @Override
    public ConfigProto getExtraConfig()
    {
        return extraConfig;
    }

    @Override
    public Fluxmap readSingleFlux(FluxReadParameters parameters)
    {
        return null;
    }
}
